### DISCUSSION TITLE
curl to http://www.google.com returns 200 but returns 307 on browser

### DISCUSSION THREAD

**Discussion by smc181002** [curl/curl#7629] [Q&A] [answered]
when recorded the requests on chrome dev tools, first response was 307 response and after the redirect lead to a 200 response
![image](https://user-images.githubusercontent.com/53662575/130832611-5663e536-6b38-4260-bd0c-812aee591334.png)
But in the case of curl, it directly returned 200
![image](https://user-images.githubusercontent.com/53662575/130832634-3265ab12-b096-43f1-b054-8f48e25d9021.png)

I think it is because this 307 is a [temporary redirect](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307) and the response we get from curl is the final response after the redirect
i just want to confirm this 
thank you

**Comment by dfandrich** [comment, marked as answer]
Questions are better asked on the curl mailing list rather than in a Github
issue.

On Wed, Aug 25, 2021 at 09:59:53AM -0700, Meher Chaitanya wrote:
> I think it is because this 307 is a temporary redirect and the response we get
> from curl is the final response after the redirect

curl won't follow redirects without the -L option. It could be Google is
looking at the User-Agent to decide whether to redirect to a more appropriate
site depending on the browser, or it could be the browser internally
redirecting to the https: version of the site because of HSTS. Enabling HSTS
with curl should give similar results.
